kernel/runtime/cxx/runtime/cxxsupport.cc

```diff
@@ -49,23 +49,66 @@
 #include "runtime/PageMap.hh"
 #include "runtime/KernelMemory.hh"
 #include "runtime/SimpleCapAlloc.hh"
+#include "runtime/ProcessorAllocator.hh"
 #include "runtime/tls.hh"
 #include "runtime/futex.hh"
 #include "runtime/umem.hh"
+#include "runtime/thread-extra.hh"
 
 extern mythos::InvocationBuf* msg_ptr asm("msg_ptr");
 extern mythos::Portal portal;
 extern mythos::CapMap myCS;
 extern mythos::PageMap myAS;
 extern mythos::KernelMemory kmem;
-extern mythos::SimpleCapAllocDel capAlloc;
+extern mythos::SimpleCapAlloc< mythos::init::APP_CAP_START
+  , mythos::init::SIZE-mythos::init::APP_CAP_START> capAlloc;
+extern mythos::ProcessorAllocator pa;
 
 #ifndef NUM_CPUS
 #define NUM_CPUS (2)
 #endif
 
 #define PS_PER_TSC (0x0000000000000181)
 
+struct PthreadCleanerSemaphore{
```

Comment:
This does not seem to be a semaphore to me.

Reply:
renamed